### ISSUE TITLE
Updates to the location marker indicators in search results

### ISSUE THREAD

**Issue by laflannery** [department-of-veterans-affairs/va.gov-cms#12906]
## Description
In order to make the location markers more useful and clearer to screen reader users, there are a couple improvements that could be done to the current design:

1. Use numbers instead of letters for the markers:
   - When working with list content, numbers is a more common and expected pattern. Letters can introduce some confusion
   - If letters is a limitation of the map marker itself and cannot be changed:
     - Can we add "." after the letter in the marker to create short pause for the screen reader. This can be visually hidden using the sr-only class.  

2. Add the location marker to the heading with the title of the search result
   - If the marker and the title are wrapped into the heading tag together, this will allow the user to hear the marker associated with the title and understand what this means when on the map.
   - At a very basic level, the HTML could look like:
 `<h3><span>{map marker}</span><a href="{link}">{Facility Name}</a></h3>`

## Acceptance Criteria

- [ ] Location marker uses number if possible
- [ ] Location marker and facility heading are programmatically associated
- [ ] Accessibility lead review

**Comment by laflannery** [comment]
@thejordanwood This is one where we might need to investigate feasibility

**Comment by laflannery** [comment]
While we don't want to emulate the [GI Bill locator](https://www.va.gov/education/gi-bill-comparison-tool/?search=location&location=02210) really in anyway, their map does what I want here. The markers are associated to the results list and so when I click or use the keyboard on one of the markers I am taken to that item in the results.

**Comment by laflannery** [comment]
Closing in favor of #14912
